src/api/undeploy/helpers/github/delete-github-file.js

```diff
@@ -0,0 +1,60 @@
+import { getLatestCommitSha } from '~/src/helpers/github/get-latest-commit-sha.js'
+import { graphql } from '~/src/helpers/oktokit-graphql.js'
+
+/**
+ * @typedef {import("pino").Logger} Logger
+ * @param {string} owner
+ * @param {string} repo
+ * @param {string} branch
+ * @param {string} commitMessage
+ * @param {string} filePath
+ * @param {Logger} logger
+ */
+async function deleteGithubFile(
+  owner,
+  repo,
+  branch = `main`,
+  commitMessage,
+  filePath,
+  logger
+) {
+  let startTime = Date.now()
+  const commitSha = await getLatestCommitSha(owner, repo, branch)
+  logger.info(`getLatestCommitSha: ${Date.now() - startTime}ms`)
+
+  startTime = Date.now()
+
```

Comment:
with `graphQL` in the codebase a few helpers would be of help:
- https://www.npmjs.com/package/eslint-plugin-graphql
- https://www.npmjs.com/package/graphql-tag